packages/table/src/table.tsx

```diff
@@ -897,6 +919,22 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
         }
     }
 
+    private areChildrenSameShape(newChildArray: Array<React.ReactElement<IColumnProps>>) {
+        if (this.childrenArray.length !== newChildArray.length) {
+            return false;
+        }
+        return this.childrenArray.every((child, index) => {
+            const newChild = newChildArray[index];
+            if (child === newChild) {
+                return true;
+            }
+            if (child == null || newChild == null) {
```

Comment:
May not be likely, but what if they're both `null` or both `undefined`?

Reply:
doesn't the comparison above catch that?